sycl/source/detail/program_impl.cpp

```diff
@@ -362,7 +362,8 @@ void program_impl::create_cl_program_with_source(const std::string &Source) {
   }
 
   if (Err != PI_SUCCESS) {
-    Plugin.reportPiError(Err, "create_cl_program_with_source()");
+    std::string CallingFunction = "create_cl_program_with_source()";
+    Plugin.reportPiError(Err, CallingFunction);
```

Comment:
what's wrong with prev version?

Reply:
there were compile errors as `std::string &` requires a `lvalue`.

Reply:
Maybe use std::string_view?

Reply:
it is kind of ugly that you cannot pass C-string directly, either use string_view or just "const char*"

Reply:
agree. how about using `std::string` instead of `std::string &`? that will resolve the lvalue issue.

Reply:
This way there would be a copy made during the call. How about using "std::string &" but passing as string("bla-bla-bla")

Reply:
this didn't work either. i am using `const char*`